src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -612,10 +619,24 @@ fn add_playpen_pre(html: &str, playpen_config: &Playpen) -> String {
                 if (!classes.contains("ignore") && !classes.contains("noplaypen"))
                     || classes.contains("mdbook-runnable")
                 {
+                    let contains_e2015 = classes.contains("edition2015");
+                    let contains_e2018 = classes.contains("edition2018");
+                    let edition_class = if contains_e2015 || contains_e2018 {
+                        // the user forced edition, we should not overwrite it
```

Comment:
This allow users to write ` ```rust,edition2015` even with book.edition=2018 and their code will be sent to playpen with edition2015. I don't know how to implement the same behavior in `mdbook test`.

Reply:
`mdbook test` with a nightly toolchain honors the `edition2015` annotation.  1.41 will honor it as of https://github.com/rust-lang/rust/pull/66238.

Reply:
Will it resolve in an error if there will be `edition=2015` in bool.toml and ` ```edition2018` in the block of code?

Reply:
No, the inline annotation overrides the global default.